### DISCUSSION TITLE
Configure Index Backend and Tracker from Frontend

### DISCUSSION THREAD

**Discussion by BelieveInBunny** [torrust/torrust-index-gui#94] [Feature requests]
(edit @da2ce7)

- [ ] Prompt for creating a new administrator account, if missing.
- [ ] Prompt to review and change the backend and tracker configuration on first load.

(original: @BelieveInBunny ):
Torrust needs at least one administrator account. On first boot it should show an install wizard to create an admin account and to set the tracker/website config.

**Comment by josecelano** [comment]
> (edit @da2ce7)
> 
> * [ ]  Prompt for creating a new administrator account, if missing.
> * [ ]  Prompt to review and change the backend and tracker configuration on first load.
> 
> (original: @BelieveInBunny ): Torrust needs at least one administrator account. On first boot it should show an install wizard to create an admin account and to set the tracker/website config.

Hi @BelieveInBunny , right now, this is what's implemented:

- The first registered used is an amdin
- There is only one admin
- There are only two roles: `admin` and the `standard` user. And `guests` (unauthenticated users).

Regarding the tracker and website config, you can change the settings on the admin panel:

![admin-settings](https://github.com/torrust/torrust-index-frontend/assets/58816/a5b2c706-f4b9-4f3a-8c6f-80afe645bd01)

But it's something I do not like because of the following:

- https://github.com/torrust/torrust-index-backend/issues/161
- https://github.com/torrust/torrust-index-backend/issues/144

I would add a new table, `settings` with all the configuration the admin can update.

Regarding changing the tracker configuration from the frontend, first we should decide which is the role of the tracker. See https://github.com/torrust/torrust-index-backend/discussions/186

Is the tracker only an associated/linked tracker to the index, or it's an independent service
e you can connect to from the index? Are we going to have more than one tracker in the future?

I think we have to answer all those questions before implementing a solution to make installation and configuration easier.

**Comment by josecelano** [comment]
Hi @BelieveInBunny, there have been a lot of changes. Now, the configuration is read-only in the admin panel. The main reasons are:

- Inconsistencies injecting configuration via env vars
- Allow to run the app with config file

Automatic deployments, containerization, and alternative infrastructure (cloud, ...) are sometimes more complex to implement with that feature.

Regarding installation, we have opened a new issue: https://github.com/torrust/torrust-index/issues/775

We will open new issues regarding the ability to change the configuration on the fly (at runtime). There are two types of configuration:

- Static: needed to start the service
- Dynamic: things you can change on the fly

Only the database connection would be needed to start the service. The rest could be moved into the database. However, moving some options to the database would require changing a lot of code, and they probably don't make much sense. For example, running a new UDP tracker on a new port would be possible. We don't intend to reach that level, but we want to move most of the CMS configuration to the database. For example, the site name, demo warnings, etc. I will create a new issue to move those values to the database, so you will be able to change some settings again, but not all.

If you could tell us which options you wanted to dynamically update from the front end, it would help us define the priorities about what to move first.
